src/lib/eventListeners.js

```diff
@@ -31,24 +32,46 @@ export default {
       componentsMap.set(component, components)
     }
 
-    components.add(cb)
+    components.add({ cb, priority })
+    cache.delete(event) // Invalidate the cache when a new callback is added
   },
+
   executeListeners(event, params) {
     const componentsMap = eventsMap.get(event)
-    if (componentsMap) {
-      componentsMap.forEach((component) => {
-        component.forEach((cb) => {
-          cb(params)
-        })
+    if (!componentsMap) {
+      return true // No listeners so execution can be considered successful
+    }
+
+    if (!cache.has(event)) {
+      // collect all callbacks and sort them by priority
+      const allCallbacks = []
+      componentsMap.forEach((components, component) => {
+        components.forEach((callbackObj) => allCallbacks.push({ ...callbackObj, component }))
       })
+      allCallbacks.sort((a, b) => b.priority - a.priority)
+      cache.set(event, allCallbacks) // Cache the sorted callbacks with component context
+    }
+
+    for (const { cb, component } of cache.get(event)) {
+      const result = cb.call(component, params)
+      if (result === false) {
+        return false // Stop propagation if any listener returns false
+      }
     }
+
+    return true // All listeners executed without stopping propagation
   },
+
   removeListeners(component) {
-    eventsMap.forEach((componentMap) => {
-      const cmp = componentMap.get(component)
-      if (cmp) {
-        cmp.clear()
-        componentMap.delete(cmp)
+    eventsMap.forEach((componentsMap, event) => {
```

Comment:
good moment to also refactor this `forEach` to a `while` loop

Reply:
used `for...of` instead of `forEach`.